jobs/data_job.go

```diff
@@ -20,9 +20,29 @@ func DataSourceJob() *schema.Resource {
 		var list []Job
 		var err error
 		if data.Name != "" {
-			list, err = jobsAPI.ListByName(data.Name, false)
+			list, err = jobsAPI.ListByName(data.Name, true)
 		} else {
+			// if name is not provided, need to list all the jobs and search by id
+			// does not expand tasks to limit the results size
 			list, err = jobsAPI.List()
```

Comment:
If we have a job ID, then why not simply call `Read` using that ID? It will be more efficient :-)

Reply:
this is a lot simpler now